pkg/queue/cluster_queue_impl.go

```diff
@@ -210,6 +210,24 @@ func (c *clusterQueueBase) Dump() (sets.Set[string], bool) {
 	return elements, true
 }
 
+func (c *clusterQueueBase) Snapshot(maxCount int32) ([]*kueue.Workload, bool) {
+	if c.heap.Len() == 0 || maxCount == 0 {
+		return nil, false
+	}
+	elements := make([]*kueue.Workload, c.heap.Len())
+	for i, e := range c.heap.List() {
```

Comment:
Does this list come sorted?

Reply:
I'm not sure, because it's a map `items map[string]*heapItem`
I guess I need to sort according to the keys in the heap

Reply:
No, it should be sorted according to the sorting rules of the queue
https://github.com/kubernetes-sigs/kueue/blob/0aab54ad4a1707e5b3aa235604eb99dfb3422705/pkg/queue/cluster_queue_strict_fifo.go#L47

Ultimately, what we want is the elements that are top N spots in the queue.

Reply:
added sort before calling `List()`
https://github.com/kubernetes-sigs/kueue/pull/1069/files#diff-2495a98a0a5ba90a95278217726f61a78f1d65943ea302129d2d9e8697306847R175